lib/metalsmith.js

```diff
@@ -22,12 +23,42 @@ const views = [
   paths.govukfrontend
 ]
 
+const coloursAppliedSCSS = scssToJson('./node_modules/@govuk-frontend/globals/settings/_colours-applied.scss', {
+  dependencies: [{path: './node_modules/@govuk-frontend/globals/settings/_colours-palette.scss'}]
+})
+
+const coloursPaletteSCSS = scssToJson('./node_modules/@govuk-frontend/globals/settings/_colours-palette.scss')
+
+// combine both SCSS data objects - some Main colours are only in Palette
+
+const coloursSCSS = Object.assign(coloursAppliedSCSS, coloursPaletteSCSS)
+
+// process colours data
+
+var colours = require('../data/colours.json')
+
+for (var groupName in colours.main){
```

Comment:
Check if `colours.main`, `colours.greyscale`, `colours.extended` exist before loop

Reply:
interesting 0 this line will fail silently if main is missing, but the other lines (47, 51) that use 'for ... of' fail with an error.

I think failing is what we need, so I'll do a check like you say and exit with an error if they don't exist.